### PR TITLE
 [refactor]  Minors numeric improvements

### PR DESCRIPTION
**Pull request by arturobernalg**
* Simplify pointless arithmetic expression.
* Integer multiplication  cast to long.
* Use valueOf
* Simplify pointless arithmetic expression.

### CODE REVIEW

**Comment by arturobernalg** [comment]
> @arturobernalg see this comment [#118 (comment)](https://github.com/esa/nanosat-mo-framework/pull/118#issuecomment-951703698)
Hi @TanguySoto 
changed.
TY

**Comment by dmarszk** [comment]
I would not simplify pointless operations like 0*4, 1*4 - these are there to make the code more readable when defining a number of indexes/offsets, and the compiler optimises it out anyway. As for other changes - they look good to me. Please rebase on top of the latest dev.

**Comment by arturobernalg** [comment]
> I would not simplify pointless operations like 0_4, 1_4 - these are there to make the code more readable when defining a number of indexes/offsets, and the compiler optimises it out anyway. As for other changes - they look good to me. Please rebase on top of the latest dev.

Ok, I already rebase from master and revert the pointless simplifications.

File: mission/simulator/platform-services-impl/src/main/java/esa/mo/platform/impl/provider/softsim/CameraSoftSimAdapter.java [hunk 1]
```diff
@@ -155,7 +155,7 @@ private byte[] convertImage(byte[] rawImage, final PictureFormat targetFormat) t
       ret = new byte[rgba.length * 3];
       for (int i = 0; i < rgba.length; ++i) {
         final int pixelval = rgba[i];
-        ret[i * 3 + 0] = (byte) ((pixelval >> 16) & 0xFF); // R
+        ret[i * 3] = (byte) ((pixelval >> 16) & 0xFF); // R
```
Comment (dmarszk) [inline mission/simulator/platform-services-impl/src/main/java/esa/mo/platform/impl/provider/softsim/CameraSoftSimAdapter.java:158]: Do not change this line.

File: sdk/tools/consumer-test-tool/src/main/java/esa/mo/nmf/ctt/services/mc/ParameterPublishedValues.java [hunk 2]
```diff
@@ -133,8 +133,8 @@ public void monitorValueNotifyReceived(final MALMessageHeader msgHeader,
                         String convertedValue = HelperAttributes.attribute2string(parameterValue.getConvertedValue());
 
                         boolean isNotValid = (validityState.getValue() != ValidityState._VALID_INDEX);
-                        labels[index + 0 * numberOfColumns].setNewValue(nameId, isNotValid);
-                        labels[index + 1 * numberOfColumns].setNewValue(validity, isNotValid);
+                        labels[index].setNewValue(nameId, isNotValid);
```
Comment (dmarszk) [inline sdk/tools/consumer-test-tool/src/main/java/esa/mo/nmf/ctt/services/mc/ParameterPublishedValues.java:136]: Do not change these lines.

**Review by dmarszk — CHANGES_REQUESTED** [review]

File: mission/simulator/platform-services-impl/src/main/java/esa/mo/platform/impl/provider/softsim/CameraSoftSimAdapter.java (hunk 1 above)
Comment (arturobernalg) [reply mission/simulator/platform-services-impl/src/main/java/esa/mo/platform/impl/provider/softsim/CameraSoftSimAdapter.java]: done

File: sdk/tools/consumer-test-tool/src/main/java/esa/mo/nmf/ctt/services/mc/ParameterPublishedValues.java [hunk 3]
```diff
@@ -77,8 +77,8 @@ public ParameterPublishedValues(final ParameterConsumerServiceImpl parameterServ
         }
 
         // Info
-        labels[0 * numberOfColumns].setNewValue("Obj Instance Id", false);
-        labels[1 * numberOfColumns].setNewValue("Validity State", false);
+        labels[0].setNewValue("Obj Instance Id", false);
```
[the remark this answers is not in the document]
Comment (arturobernalg) [reply sdk/tools/consumer-test-tool/src/main/java/esa/mo/nmf/ctt/services/mc/ParameterPublishedValues.java]: done

File: sdk/tools/consumer-test-tool/src/main/java/esa/mo/nmf/ctt/services/mc/ParameterPublishedValues.java (hunk 2 above)
Comment (arturobernalg) [reply sdk/tools/consumer-test-tool/src/main/java/esa/mo/nmf/ctt/services/mc/ParameterPublishedValues.java]: done

File: mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/peripherals/PFineADCS.java [hunk 4]
```diff
@@ -66,8 +66,8 @@ public static class FWRefFineADCS {
         //Below are indexes to the fields in the resulting byte arrays
         public static class SENSORTM_IDX{
             //Byte offset
-            public final static int MAG_FIELD_X                      = 0*4;
-            public final static int MAG_FIELD_Y                      = 1*4;
+            public final static int MAG_FIELD_X                      = 0;
```
[the remark this answers is not in the document]
Comment (arturobernalg) [reply mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/peripherals/PFineADCS.java]: done

**Pull request closed without merging**